.evergreen/config.yml

```diff
@@ -16,33 +43,146 @@ functions:
         add_to_path:
           - .evergreen
         script: |
+          ${PREPARE_SHELL}
           ${script}
 
+  "prepare resources":
+    - command: subprocess.exec
+      params:
+        working_dir: src
+        binary: bash
+        add_to_path:
+          - .evergreen
+        args:
+          - fetch-drivers-tools.sh
+        include_expansions_in_env:
+          - DRIVERS_TOOLS
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          echo "{ \"releases\": { \"default\": \"$MONGODB_BINARIES\" }}" > $MONGO_ORCHESTRATION_HOME/orchestration.config
+
+  "bootstrap mongo-orchestration":
+    - command: subprocess.exec
+      params:
+        binary: sh
+        args:
+          - ${DRIVERS_TOOLS}/.evergreen/run-orchestration.sh
+        include_expansions_in_env:
+          - AUTH
+          - SSL
+          - TOPOLOGY
+          - REQUIRE_API_VERSION
+          - MONGODB_VERSION
+          - ORCHESTRATION_FILE
+          - MONGODB_BINARIES
+    - command: expansions.update
+      params:
+        file: mo-expansion.yml
+
+  "stop mongo orchestration":
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          cd ${MONGO_ORCHESTRATION_HOME}
+          # source the mongo-orchestration virtualenv if it exists
+          if [ -f venv/bin/activate ]; then
+            . venv/bin/activate
+          elif [ -f venv/Scripts/activate ]; then
+            . venv/Scripts/activate
+          fi
+          mongo-orchestration stop || true
+
+  "cleanup":
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          rm -rf $DRIVERS_TOOLS || true
+
+# Functions to run before all tasks (except those in task groups).
 pre:
-  - func: fetch source
+  - func: "fetch source"
+  - func: "create expansions"
+  - func: "prepare resources"
 
+# Functions to run after all tasks (except those in task groups).
 post:
+  - func: "stop mongo orchestration"
+  - func: "cleanup"
 
+#######################################
+#                Tasks                #
+#######################################
 tasks:
   - name: static-check
-    tags: ["ci"]
+    type: test
+    tags:
+      - ci
+      - static-check
     commands:
-      - func: exec script
-        vars:
+      - func: "exec script"
+        params:
           script: static-check.sh
 
   - name: run-test
```

Comment:
it is not too soon for the first PR containing testing code is about to be merged (https://github.com/mongodb/mongo-hibernate/pull/8). I will keep it in mind to test the above stuff after that happened; otherwise it is hard to verify and tweaking evergreen project to make that hapen seems troublesome.